app/client/components/consent/PrivacySettings.tsx

```diff
@@ -212,37 +226,252 @@ export class PrivacySettings extends Component<{}, State> {
               }}
               css={css`
                 ${buttonStyles};
-                background-color: ${palette.yellow.dark};
               `}
             >
-              Save and close
+              Save and continue
             </button>
           </div>
         </form>
       </div>
     );
   }
 
-  public renderPurposeItems(): React.ReactNode {
-    const purposeItemKeys = Object.keys(this.state.purposes);
-    const purposeItemCount = purposeItemKeys.length;
+  private doScrolling(query: string, duration: number): void {
+    const element: HTMLElement | null = document.querySelector(query);
+    if (!element) {
+      return;
+    }
+
+    const elementY: number =
+      window.pageYOffset + element.getBoundingClientRect().top;
+    const startingY: number = window.pageYOffset;
+    const diff: number = elementY - startingY;
+    let start: number;
 
-    return purposeItemKeys.map(
-      (key: string, index: number): React.ReactNode => {
-        const purposeId = key as PurposeType;
+    const animationStep: FrameRequestCallback = timestamp => {
+      if (!start) {
+        start = timestamp;
+      }
+      // Elapsed milliseconds since start of scrolling.
+      const time: number = timestamp - start;
+      // Get percent of completion in range [0, 1].
+      const percent: number = Math.min(time / duration, 1);
+
+      window.scrollTo(0, startingY + diff * percent);
```

Comment:
This animation looks well-written from a performance point of view, but it doesn't have any easing. Linear animations look jarring and unnatural, typically an easing function is used so the animation is non-linear. Here's some research from Google's material design (which uses cubic bezier):
https://material.io/design/motion/speed.html#controlling-speed

It's normally better to use CSS transitions. It saves all this code, you get stuff like easing for free, and browsers are really good at optimising them. Since this is doing scroll position I don't think that'll be possible though!